### PR TITLE
[#2967] Factor out shared code between PC & NPC sheets. Add item description collapsible sections.

### LINKED ISSUE #2967: Option for old Expanding Descriptions in place of Tooltips
While the tooltips are real nice, they are kind of hard to read / manage a bit. Would it be at all possible to maybe add some functionality back to the expanding item descriptions to the character sheets?

### PR DESCRIPTION
**Pull request by Fyorl**
- Closes #2967

Sorry for the size of the PR, a large portion is just code that has been moved without modification, or file renames. I will highlight the sections you can safely skip below:

- `v2/actors.less` has shared code moved from `v2/character.less`.
- `sheet-v2-mixin.mjs` has shared code moved from `character-sheet-2.mjs`
- `actor-classes.hbs` was factored out of `character-features.hbs` (now `creature-features.hbs`)
- `actor-trait-pills.hbs` was factored out of `character-details.hbs`
- All of the `column-*.hbs` files were factored out of `character-features.hbs` (now `creature-features.hbs`)
- `character-spells.hbs` was renamed to `creature-spells.hbs`

`creature-features.hbs` was renamed from `character-features.hbs` but does contain some fairly significant refactoring so is worth a look. `creature-spells.hbs`, `inventory2.hbs` and `creature-features.hbs` show a lot of changes because the item rows needed to be further wrapped in order to support the extra item descriptions, so it's worth taking a quick look at that but otherwise most changes are just whitespace.

### CODE REVIEW

File: less/elements.less [hunk 1]
```diff
@@ -114,6 +114,7 @@ item-list-controls search {
       font-weight: normal;
       padding: .5rem 1.75rem .5rem 0;
       justify-content: start;
+      text-align: left;
```
Comment (arbron) [inline less/elements.less:117]: If we wanted to use more logical properties.

```suggestion
      text-align: start;
```

File: less/v2/inventory.less [hunk 2]
```diff
@@ -464,17 +468,38 @@
         align-items: center;
       }
 
-      @container (min-width: 600px) {
-        .item-price, .item-formula, .spell-uses { display: flex; }
+      /* Item Description */
+      .item-description {
+        font-size: var(--font-size-11);
+
+        .item-summary {
```
Comment (arbron) [inline less/v2/inventory.less:475]: I've noticed that if you don't have a description then the pills get push right up against the rest of the row:

Screenshot 2024-07-04 at 09 19 27

Probably want a bit of padding there to make it a bit more even compared to the bottom padding.

File: module/applications/actor/sheet-v2-mixin.mjs [hunk 3]
```diff
@@ -0,0 +1,633 @@
...
+    async getData(options) {
+      this._concentration = this.actor.concentration; // Cache concentration so it's not called for every item.
+      const context = await super.getData(options);
+      context.editable = this.isEditable && (this._mode === this.constructor.MODES.EDIT);
+      context.cssClass = context.editable ? "editable" : this.isEditable ? "interactable" : "locked";
+      const activeTab = (game.user.isGM || !this.actor.limited) ? this._tabs?.[0]?.active ?? "details" : "biography";
+      context.cssClass += ` tab-${activeTab}`;
+      context.sidebarCollapsed = !!game.user.getFlag("dnd5e", `sheetPrefs.character.tabs.${activeTab}.collapseSidebar`);
+      if ( context.sidebarCollapsed ) context.cssClass += " collapsed";
+      const { attributes, traits } = this.actor.system;
```
Comment (arbron) [inline module/applications/actor/sheet-v2-mixin.mjs:126]: `traits` isn't currently being used.

```suggestion
      const { attributes } = this.actor.system;
```

File: module/applications/actor/sheet-v2-mixin.mjs [hunk 4]
```diff
@@ -0,0 +1,633 @@
...
+     * @protected
+     */
+    _applyReferenceTooltips(element) {
+      if ( "tooltip" in element.dataset ) return;
+      const uuid = element.dataset.referenceTooltip;
+      element.dataset.tooltip = `
+      <section class="loading" data-uuid="${uuid}"><i class="fas fa-spinner fa-spin-pulse"></i></section>
+    `;
+    }
+  }
```
Comment (arbron) [inline module/applications/actor/sheet-v2-mixin.mjs:632]:
```suggestion
  };
```

File: module/applications/actor/sheet-v2-mixin.mjs [hunk 5]
```diff
@@ -0,0 +1,633 @@
...
+    _mode = this.constructor.MODES.PLAY;
+
+    /**
+     * The cached concentration information for the character.
+     * @type {{items: Set<Item5e>, effects: Set<ActiveEffect5e>}}
+     * @internal
+     */
+    _concentration;
+
+    /* -------------------------------------------- */
```
Comment (arbron) [inline module/applications/actor/sheet-v2-mixin.mjs:49]: Might be nice to have some separators to distinguish the grouping of methods (e.g. "Rendering" & "Event Listeners & Handlers").

Would involve shifting the `activateListeners` method beneath the other preparation methods.

File: templates/actors/parts/columns/column-feature-controls.hbs [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+    {{!-- Deleting --}}
+    <a class="item-control item-action" data-action="delete" data-tooltip="DND5E.ItemDelete"
+       aria-label="{{ localize "DND5E.ItemDelete" }}">
+        <i class="fas fa-trash"></i>
+    </a>
+    {{/if}}
+
+    {{!-- Expand/Collapse --}}
+    <a class="item-control interface-only" data-toggle-description
```
Comment (arbron) [inline templates/actors/parts/columns/column-feature-controls.hbs:18]: Not hit box of the context menu button seems to overlapping the expand/collapse button making it hard to click in edit mode or on the inventory tab:

Screenshot 2024-07-04 at 09 19 01

File: module/utils.mjs [hunk 7]
```diff
@@ -469,7 +490,8 @@ export function registerHandlebarsHelpers() {
     "dnd5e-linkForUuid": (uuid, options) => linkForUuid(uuid, options.hash),
     "dnd5e-itemContext": itemContext,
     "dnd5e-numberFormat": (context, options) => formatNumber(context, options.hash),
-    "dnd5e-textFormat": formatText
+    "dnd5e-textFormat": formatText,
+    "dnd5e-formatModifier": formatModifier
```
Comment (arbron) [inline module/utils.mjs:494]: Would be nice to move this up so these are in alphabetical order (though I suppose `itemContext` is also in the wrong spot :/).

File: module/utils.mjs [hunk 8]
```diff
@@ -457,6 +466,18 @@ function concealSection(conceal, options) {
 
 /* -------------------------------------------- */
 
+/**
+ * Format a modifier for display with its sign separate.
+ * @param {number} mod  The modifier.
+ * @returns {Handlebars.SafeString}
+ */
+function formatModifier(mod) {
```
Comment (arbron) [inline module/utils.mjs:474]: This should be up in the "Formatters" grouping new the top of the file.

File: module/utils.mjs [hunk 9]
```diff
@@ -301,17 +301,26 @@ export async function preloadHandlebarsTemplates() {
...
     "systems/dnd5e/templates/actors/parts/actor-features.hbs",
     "systems/dnd5e/templates/actors/parts/actor-spellbook.hbs",
     "systems/dnd5e/templates/actors/parts/actor-warnings.hbs",
+    "systems/dnd5e/templates/actors/parts/actor-classes.hbs",
     "systems/dnd5e/templates/actors/tabs/character-details.hbs",
-    "systems/dnd5e/templates/actors/tabs/character-features.hbs",
-    "systems/dnd5e/templates/actors/tabs/character-spells.hbs",
+    "systems/dnd5e/templates/actors/tabs/creature-features.hbs",
+    "systems/dnd5e/templates/actors/tabs/creature-spells.hbs",
     "systems/dnd5e/templates/actors/tabs/character-biography.hbs",
```
Comment (arbron) [inline module/utils.mjs:314]:
```suggestion
    "systems/dnd5e/templates/actors/tabs/character-biography.hbs",
    "systems/dnd5e/templates/actors/tabs/creature-features.hbs",
    "systems/dnd5e/templates/actors/tabs/creature-spells.hbs",
```

**Review by arbron — CHANGES_REQUESTED** [review]
Looking good.

**Comment by Fyorl** [comment]
Thank you for the feedback, I have addressed it in the latest commit. There are also some additional changes in that commit from more code sharing I factored out today for the biography tab.

**Review by arbron — APPROVED** [review]

**Pull request merged**